syntax = "proto3";

message TaskQueryReq // IRequest
{
    int32 RpcId = 90;
    //    int32 gid = 1;  // gid 没有必要，客户端没有要查询其他玩家进度信息的需求
}

message TaskQueryRsp // IResponse
{
    enum ErrorCode {
        SUCCEED = 0;
        UNKOWN_EORROR = -1;
    }
    int32 RpcId = 90;
    int32 Error = 91;
    string Message = 92;

    repeated int32 done_tasks = 1; // 现在已经完成的任务
    repeated int32 get_tasks = 2; // 现在接取到的任务
    int32 position_x = 3;
    int32 position_y = 4;
}

message TaskUpdateReq // IRequest
{
    int32 RpcId = 90;
    repeated int32 finished_task_ids = 1;
    repeated int32 get_task_ids = 2;
    int32 position_x = 3;
    int32 position_y = 4;
}

message TaskUpdateRsp // IResponse
{
    enum ErrorCode {
        SUCCEED = 0;
        UNKOWN_EORROR = -1;
    }
    int32 RpcId = 90;
    int32 Error = 91;
    string Message = 92;
}